src/tiledb/cloud/utilities/consolidate.py

```diff
@@ -0,0 +1,248 @@
+    """
+    Get a list of fragment info objects, optionally grouping fragments that have the
+    same value for the first dimension.
+
+    :param array_uri: array URI
+    :param config: config dictionary, defaults to None
+    :param group_first_dim: group by first dimension, defaults to True
+    :return: list of lists of fragment info objects
+    """
+
+    logger = get_logger()
+    logger.info("Grouping fragments for array %r", array_uri)
+
+    with tiledb.scope_ctx(config):
+        fragment_info_lists = defaultdict(list)
+
+        fis = tiledb.FragmentInfoList(array_uri)
+
+        for fi in fis:
+            if group_first_dim:
+                fragment_info_lists[fi.nonempty_domain[0]].append(fi)
+            else:
+                fragment_info_lists["all"].append(fi)
+
+        # Create a list of lists of fragment info objects.
+        results = [x for x in list(fragment_info_lists.values())]
+
+        logger.info(f"{len(fis)} fragments grouped into {len(results)} groups")
+        logger.info(f"max memory usage: {max_memory_usage() / (1 << 30):.3f} GiB")
+
+        return results
+
+
+def consolidate(
+    array_uri: str,
+    fragments: Sequence[tiledb.FragmentInfo],
+    *,
+    config: Optional[Mapping[str, Any]] = None,
+    max_fragment_size: int = MAX_FRAGMENT_SIZE,
+) -> None:
+    """
+    Consolidate fragments
+
+    :param array_uri: array URI
+    :param fragments: list of fragments
+    :param config: config dictionary, defaults to None
+    :param max_fragment_size: max size of consolidated fragments,
+        defaults to MAX_FRAGMENT_SIZE
+    """
+
+    logger = get_logger()
+    logger.info(f"Consolidating {len(fragments)} fragments")
+
+    config = tiledb.Config(config)
+    config["sm.consolidation.mode"] = "fragments"
+    config["sm.consolidation.max_fragment_size"] = max_fragment_size
+
+    # Consolidate fragments.
+    fragment_names = [basename(fi.uri) for fi in fragments]
+    with tiledb.open(array_uri, "w", config=config) as array:
+        array.consolidate(fragment_uris=fragment_names)
+
+    logger.info(f"max memory usage: {max_memory_usage() / (1 << 30):.3f} GiB")
+
+
+def convac(
+    array_uri: str,
+    *,
+    config: Optional[Mapping[str, Any]] = None,
+    vacuum_fragments: bool = False,
+) -> None:
+    """
+    Consolidate and vacuum commits and fragment metadata, with an option to
+    vacuum fragments as the first step.
+
+    :param array_uri: array URI
+    :param config: config dictionary, defaults to None
+    :param vacuum_fragments: vacuum fragments first, defaults to False
+    """
+
+    logger = get_logger()
+
+    with tiledb.scope_ctx(config):
+        if vacuum_fragments:
+            try:
+                logger.info("Vacuuming fragments")
+                tiledb.vacuum(
+                    array_uri,
+                    config=tiledb.Config({"sm.vacuum.mode": "fragments"}),
+                )
+            except:
+                raise
```

Comment:
this `try/except` is a do-nothing; was there intended to be more going on here?

Reply:
Thanks, another case of reusing code that can be simplified.